web_widget_heatmap/static/src/js/heatmap_widget.js

```diff
@@ -0,0 +1,194 @@
+        generate_element_options: function(elements) {
+            var start_date = elements.length ? elements[0].create_date.toDate() : new Date();
+            var timestamps = elements.map(function(el) {
+                return el.create_date.unix();
+            });
+            var domain = "day";
+            var range = 16;
+            var end_date = elements.length ? elements[elements.length - 1].create_date.toDate() : null;
+            var max_date = this.get_max_date({
+                start_date: start_date,
+                end_date: end_date,
+                domain: domain,
+                range: range
+            });
+            var controls = false;
+            if (max_date < end_date) {
+                controls = true;
+            }
+            return {
+                start: start_date,
+                data: _.chain(timestamps).countBy().value(),
+                dataType: 'json',
+                minDate: start_date,
+                maxDate: max_date,
+                controls: controls,
+                range: range,
+                domain: domain,
+                subDomain: "hour",
+                domainGutter: 0,
+                highlight: "now",
+                onClick: this.onClickHeatMap.bind(this),
+                label: {
+                    position: "top"
+                }
+            };
+        },
+
+        /**
+         * Render the view
+         * @private
+         */
+        _render: function () {
+            var self = this;
+            var elements = this.value ? _.pluck(this.value.data, 'data') : [];
+            var options = this.generate_element_options(elements);
+            this.controls = options.controls;
+            this.renderElement();
+            var nodeOptions = this.nodeOptions || {};
+            this.heatmap_options = _.extend(options, {itemSelector: this.$el.find('.o_field_heatmap')[0]});
+            _.each(nodeOptions, function(value, key) {
+                self.heatmap_options[key] = value;
+            });
+            this.heatmap = new CalHeatMap();
+            this.heatmap.init(this.heatmap_options);
+        },
+
+        /**
+         * Renders the element and add new events.
+         */
+        renderElement: function() {
+            this._super();
+            this.$el.find('.next').click(this.nextHeatMap.bind(this));
+            this.$el.find('.previous').click(this.previousHeatMap.bind(this));
+        },
+
+        /**
+         * Show next heatmap grid
+         */
+        nextHeatMap: function() {
+            this.heatmap.next();
+        },
+
+        /**
+         * Show previous heatmap grid
+         */
+        previousHeatMap: function() {
+            this.heatmap.previous();
+        },
+
+        /**
+         * Find the records that match the selected date and show in the tree view
+         * @param {Date} date
+         * @returns {jQuery.Deferred} Action loaded
+         */
+        onClickHeatMap: function(date) {
```

Comment:
I am not sure I understand this function, and the references to dates.

A heatmap's elements as I understand can represent a lot of things, not just dates/time. This heatmap could have as elements the records of the XMany field and depending on a field on those records proper opacity can be generated. 
I thin